include/fixed_containers/reflection.hpp

```diff
@@ -234,4 +235,31 @@ constexpr auto field_info_of()
     return field_info_of<RECURSION_TYPE, FIELD_COUNT, std::decay_t<T>>(std::decay_t<T>{});
 }
 
+namespace caching
+{
+template <typename T>
+    requires(Reflectable<std::decay_t<T>>)
+inline constexpr auto FIELD_INFO =
+    field_info_of<fixed_containers::reflection_detail::RecursionType::NON_RECURSIVE, T>();
+
+template <typename T>
+    requires(Reflectable<std::decay_t<T>>)
+inline constexpr auto FIELD_INFO_RECURSIVE_DEPTH_FIRST_ORDER =
+    field_info_of<fixed_containers::reflection_detail::RecursionType::RECURSIVE_DEPTH_FIRST_ORDER,
+                  T>();
+}  // namespace caching
+
+template <typename T, typename Func>
+    requires(Reflectable<std::decay_t<T>>)
+constexpr void for_each_field(T&& instance, Func&& func)
+{
+    constexpr const auto& FIELD_INFO = caching::FIELD_INFO<T>;
+    auto tuple_view = tuples::as_tuple_view<FIELD_INFO.size()>(instance);
+    tuples::for_each_entry(tuple_view,
+                           [&func]<typename Field>(Field&& field, std::size_t index) {
+                               std::forward<Func>(func)(FIELD_INFO.at(index).field_name(),
+                                                        std::forward<Field>(field));
```

Comment:
Maybe the order here should be flipped, i.e. Have the field reference first, then the field_name.

Reply:
Ok, leaving as is.